### PR TITLE
Fixed order by clause in the get nearby rides API

### PR DESCRIPTION
**Pull request by nikhilleo10**
### Ticket Link
Fixed order by in Get Nearby Rides API. 
We can sort with rides on top of distance.
---

### Related Links

---

### Description
Fixed order by in Get Nearby Rides API. 
We can sort with rides on top of distance.
---

### Steps to Reproduce / Test

---

---

### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---

### CODE REVIEW

File: server/gql/models/vehicles/index.js [hunk 1]
```diff
@@ -121,3 +122,16 @@ export const vehicleMutations = {
   type: GraphQLVehicle,
   model: db.vehicleModel
 };
+
+function _formatForOrderByClause(args) {
```
Comment (christin-wednesday) [inline server/gql/models/vehicles/index.js:126]: This should be reusable and unit tested.

**Review by christin-wednesday — CHANGES_REQUESTED** [review]

File: server/gql/models/vehicles/index.js (hunk 1 above)
Comment (nikhilleo10) [reply server/gql/models/vehicles/index.js:126]: I am adding this in the other file and will cover the Unit test in next PR.

File: server/gql/models/vehicles/index.js [hunk 2]
```diff
@@ -121,3 +122,16 @@ export const vehicleMutations = {
...
+  const arr = args.split(',');
+  const orderByClause = [];
+  arr.forEach(element => {
+    const queryArgs = element.trim().split(' ');
+    const orderByCol = queryArgs[0];
+    const orderByType = queryArgs[1];
+    const query = 'queryColumn'.replace('queryColumn', orderByCol);
```
Comment (christin-wednesday) [inline server/gql/models/vehicles/index.js:133]: What does this do ?

Comment (christin-wednesday) [reply server/gql/models/vehicles/index.js:133]: I mean

```
 const query = 'queryColumn'.replace('queryColumn', orderByCol);
```

Comment (nikhilleo10) [reply server/gql/models/vehicles/index.js:133]: The query is for generating literal query for Sequelize function, so I am using quotes on queryColum as the query requires the quotes to work. So I am replacing 'queryColumn' with something like `driver.user.distance_in_km`. 

This creates a query like : ORDER BY '`driver.user.distance_in_km`' ASC in SQL.

File: server/gql/models/vehicles/index.js [hunk 3]
```diff
@@ -121,3 +122,16 @@ export const vehicleMutations = {
...
+    orderByClause.push([db.sequelize.literal(query), `${orderByType}`]);
+  });
+  return orderByClause;
```
Comment (christin-wednesday) [inline server/gql/models/vehicles/index.js:136]:
```suggestion
  return orderClauses.split(',').map(orderClause => {
    const queryArgs = orderClause.trim().split(' ');
    
    if(!queryArgs.length) {
    // handle incorrect clause
    }
    
    const [orderByCol, orderByType] = queryArgs
    return [db.sequelize.literal(query), `${orderByType}`]l
  });
```

**Review by aseer-ws — APPROVED** [review]

**Pull request merged**